### DISCUSSION TITLE
Require Where Clause in Update and Delete Not Working

### DISCUSSION THREAD

**Discussion by Dark-Akkula** [supabase/supabase#20913] [Questions]
pg-safeupdate access denied?

Does anyone know how to make this required in the database?  I tried to use the alter function but I get access denied

ERROR:  42501: permission denied to set parameter "session_preload_libraries"

I ran this:

alter database postgres set session_preload_libraries = 'safeupdate';

Is my database name correct?

https://supabase.com/docs/guides/database/extensions/pg-safeupdate#enable-the-extension

**Comment by Dark-Akkula** [comment]
I updated the title to hopefully get someone to review it or make suggestions.

**Comment by GaryAustin1** [comment]
I'm curious as by default this is supposed to be installed and enabled on postgres database in Supabase instances.
Was it not for you?

**Comment by Dark-Akkula** [comment]
Hey Gary...hmm....I have had this instance for over a year and I am finally getting serious and going to use it for production soon so maybe it is because this is an older instance.  

In any event, I created a test table and ran the following and got a success message

delete from public."Test"

Do I need to request the team add this on my behalf once I upgrade or is this an issue on all instances?

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/f4024702-2508-4933-a17f-16a92a4bc0a4)
From the API it is enforced.
From the SQL editor or my IDE over the database port I now get a warning about the operation being destructive and can override and proceed.
I'm pretty sure last time a checked (a year ago maybe) that was just blocked like the API request, but could be wrong.

So at this point I bow out as I'm not sure the current state, except it appears to be installed and working for me for the API on an about year old instance.
